packages/lesswrong/server/analytics/analyticsCron.ts

```diff
@@ -0,0 +1,115 @@
+/* eslint-disable no-console */
+import moment from "moment";
+import { addCronJob } from "../cronUtil";
+import { Globals } from "../vulcan-lib";
+import PostViewsRepo from "../repos/PostViewsRepo";
+import PostViewTimesRepo from "../repos/PostViewTimesRepo";
+import IncrementalViewRepo from "../repos/IncrementalViewRepo";
+import { isEAForum } from "../../lib/instanceSettings";
+
+async function updateDailyAnalyticsCollection<N extends CollectionNameString>({
+  repo,
+  earliestStartDate,
+  latestEndDate,
+}: {
+  repo: IncrementalViewRepo<N>;
+  earliestStartDate: Date;
+  latestEndDate: Date;
+}) {
+  const { earliestWindowStart, latestWindowEnd } = await repo.getDateBounds()
```

Comment:
In this and below, there is a subtle point that we _do_ always recalculate the most recent day, because it may have had partial data. Agree this comment is useful, I'll just reword it to make sure this is clear